### ISSUE TITLE
DPL-919 Transfer PBMC pools to 10X chip

### ISSUE THREAD

**Issue by KatyTaylor** [sanger/limber#1392]
**User story**
As a user of the scRNA Core pipeline, I would like LIMS support for the transfer from the plate containing pools of PBMCs plate to the 10X chip.

**Who are the primary contacts for this story**
Katy, Abby, Lesley

**Who is the nominated tester for UAT**
Lesley, Abby

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] Rename existing purposes, along with associated pipeline and purpose config and any other references, from 'HT' to 'GEM-X' (some downstream purposes are also covered by this other story - https://github.com/sanger/limber/issues/1444):
    - [x] LRC HT 5p Chip --> LRC GEM-X 5p Chip
    - [x] LRC HT 5p GEMs --> LRC GEM-X 5p GEMs
    - [x] LRC HT 5p cDNA PCR --> LRC GEM-X 5p cDNA PCR
    - [x] LRC HT 5p cDNA PCR XP --> LRC GEM-X 5p cDNA PCR XP
- [x] I can transfer from the "LRC PBMC Pools" OR "LRC PBMC Pools Input" plate to the "LRC GEM-X 5p Chip" chip
- [x] Bed verification:
    - [x] Robot: Hamilton STARlet
    - [x] Source plate on bed 8
    - [x] Destination chip on bed 15
- [x] If there are more than 8 pools on the source plate (there shouldn't be, because the pools are determined by the LIMS), then show an error.
- [x] Integration Suite test is updated to include this transfer.

**Dependencies**
Blocked by 
- https://github.com/sanger/sequencescape/issues/3936
- https://github.com/sanger/limber/issues/1705

**Additional information**

Format for chip - 1 rows of 8 - row and column naming TBC
There are additional wells on the chip that aren't tracked - they are for reagents to be loaded. There will never be samples in these locations so don't need to be captured in LIMS.

**Comment by ac38** [comment]
We have scanned a couple of HT chips to confirm the barcode layout and it is as above, you get the part no. first which should be the same for every HT chip (2000418) followed by a unique ID after a carriage return that could have '0's at the front.
![10X HT Chip barcodes](https://github.com/sanger/limber/assets/101636077/c9b91f7d-be20-45b5-b90c-95bf1947dff1)

**Comment by KatyTaylor** [comment]
**Info about barcodes taken out of description:**

_Code written on old 'PSD annotation' diagram:_
```
require './lib/plate_map_generation'
PlateMapGeneration.generate!
```
and branch: https://github.com/sanger/sequencescape/compare/DPL-687-research 

_From old 'PSD annotation' diagram:_
![Screenshot 2023-09-27 at 11 00 52](https://github.com/sanger/limber/assets/10407809/d242a5a6-c2c0-4bdc-91c9-61212d6b9b94)

**Outstanding questions for stakeholders**
1. Above screenshot says that barcodes are 2D and might have carriage returns in them - is this still the case? Could be a big problem! _(answered above)_

**Comment by KatyTaylor** [comment]
We agreed in the meeting 21/11/23 that if the barcodes presented a big problem, they could work around it in the lab for the time being (by printing out a normal plate barcode and sticking it somewhere, perhaps on the 'holder' rather than the chip. Therefore I will not create a story to cope with these barcodes for now.

**Comment by KatyTaylor** [comment]
Updated description in anticipation of GEM-X chip decision.

**Comment by KatyTaylor** [comment]
Removed on hold label following decision to go with GEM-X chip.

**Comment by yoldas** [comment]
- [ ] Rename LRC HT 5p cDNA Input --> LRC GEM-X 5p cDNA Input

**Comment by yoldas** [comment]
We have added an update to this story to add version suffixes to the library type names. The Integration Suite branch was [here](https://gitlab.internal.sanger.ac.uk/psd/integration-suite/-/merge_requests/135). The update PRs for Limber, SS and Int. Suite have been merged.
